code/__DEFINES/milla.dm

```diff
@@ -0,0 +1,139 @@
+	if(!isnull(innate_heat_capacity) && !isnum(innate_heat_capacity))
+		CRASH("Called set_tile_atmos with non-number innate_heat_capacity.")
+	return REAL_SET_TILE_ATMOS(x, y, z, blocked_north, blocked_east, blocked_south, blocked_west, atmos_mode, external_temperature, oxygen, carbon_dioxide, nitrogen, toxins, sleeping_agent, agent_b, temperature, innate_heat_capacity)
+
+#define REAL_GET_TILE_ATMOS(args...) MILLA_CALL(get_tile_atmos, args)
+
+/proc/get_tile_atmos(x, y, z)
+	if(!isnum(x) || !isnum(y) || !isnum(z))
+		CRASH("Called get_tile_atmos without specifying coordinates.")
+	return REAL_GET_TILE_ATMOS(x, y, z)
+
+#define REAL_SPAWN_TICK_THREAD MILLA_CALL(spawn_tick_thread)
+
+/proc/spawn_milla_tick_thread()
+	return REAL_SPAWN_TICK_THREAD
+
+#define REAL_GET_INTERESTING_TILES MILLA_CALL(get_interesting_tiles)
+
+/proc/get_interesting_atmos_tiles()
+	return REAL_GET_INTERESTING_TILES
+
+#define REAL_REDUCE_SUPERCONDUCTIVITY(args...) MILLA_CALL(reduce_superconductivity, args)
+
+/proc/reduce_superconductivity(x, y, z, list/superconductivity)
+	if(!isnum(x) || !isnum(y) || !isnum(z))
+		CRASH("Called reduce_superconductivity without specifying coordinates.")
+	var/north = superconductivity[1]
+	var/east = superconductivity[2]
+	var/south = superconductivity[3]
+	var/west = superconductivity[4]
+	if(!isnull(north) && !isnum(north))
+		CRASH("Called reduce_superconductivity with non-number north.")
+	if(!isnull(east) && !isnum(east))
+		CRASH("Called reduce_superconductivity with non-number east.")
+	if(!isnull(south) && !isnum(south))
+		CRASH("Called reduce_superconductivity with non-number south.")
+	if(!isnull(west) && !isnum(west))
+		CRASH("Called reduce_superconductivity with non-number west.")
+
+	return REAL_REDUCE_SUPERCONDUCTIVITY(x, y, z, north, east, south, west)
+
+#define REAL_RESET_SUPERCONDUCTIVITY(args...) MILLA_CALL(reset_superconductivity, args)
+
+/proc/reset_superconductivity(x, y, z)
+	if(!isnum(x) || !isnum(y) || !isnum(z))
+		CRASH("Called reset_superconductivity without specifying coordinates.")
+	return REAL_RESET_SUPERCONDUCTIVITY(x, y, z)
+
+#define REAL_SET_TILE_ATMOS_BLOCKING(args...) MILLA_CALL(set_tile_atmos_blocking, args)
+
+/proc/set_tile_atmos_blocking(x, y, z, list/blocking)
+	if(!isnum(x) || !isnum(y) || !isnum(z))
+		CRASH("Called set_tile_atmos_blocking without specifying coordinates.")
+	var/north = blocking[1]
+	var/east = blocking[2]
+	var/south = blocking[3]
+	var/west = blocking[4]
+	if(!isnull(north) && !isnum(north))
+		CRASH("Called set_tile_atmos_blocking with non-number north.")
+	if(!isnull(east) && !isnum(east))
+		CRASH("Called set_tile_atmos_blocking with non-number east.")
+	if(!isnull(south) && !isnum(south))
+		CRASH("Called set_tile_atmos_blocking with non-number south.")
+	if(!isnull(west) && !isnum(west))
+		CRASH("Called set_tile_atmos_blocking with non-number west.")
+
+	return REAL_SET_TILE_ATMOS_BLOCKING(x, y, z, north, east, south, west)
+
+#undef MILLA
```

Comment:
Should we add a get_version call like RUSTG has to make sure that we dont get mismatches between the DMAPI calls and the lib's calls?

Reply:
I'm not worried about that because MILLA is currently inside the same repository as the main code, so there's little risk of it getting out of sync. If we decide to separate that out at some point, it might be wise to do so.